### DISCUSSION TITLE
converting existing text column to translatable

### DISCUSSION THREAD

**Discussion by BenjaKemp** [spatie/laravel-translatable#259] [Q&A]
i'm trying to convert and existing text column to translatable. I find that when i add the column name to the the protected translatable array i am no longer able to access it as i did before ($model->key)

I assume that this is because its looked for a translation but can't find one. Is there a way for me to return to contents of the column? I want to retrieve the text and and replace it with a json

**Comment by cscognamiglio** [comment]
I am guessing you solved this by now, I was looking into doing the same thing.
The solution is to use the DB facade and do a raw query, you will get the results in an array of objects that then you can iterate and update your model. Below a simple example (I created a laravel command to do the conversion), you should some how first check if the field has a translation already.

```
use App\Models\Page;
use Illuminate\Support\Facades\DB;

$raw_pages = DB::select('select * from pages');

foreach($raw_pages as $raw_page) {
            echo $raw_page->title . "\n";

            $page = Page::find($raw_page->id);
            $page->setTranslation('title', 'fr', $raw_page->title);
            $page->save();
}
```

**Comment by youseffatouraee** [comment]
I would suggest using the getRawOriginal() method as a cleaner solution, like:
```php
use App\Models\Page;
$pages = Page::all();
foreach ($pages as $page) {
        $page->setTranslation('title', 'en', $page->getRawOriginal('title'))->save();
}
```
Change the 'Page' model with your desired model.
